s3/client.bal

```diff
@@ -300,16 +300,21 @@ public isolated client class Client {
         };
 
         GET|PUT httpMethod;
-        if action is CREATE || action is ObjectCreationHeaders {
+
+        if action is ObjectCreationHeaders {
+            httpMethod = PUT;
+            populateCreateObjectHeaders(requestHeaders, action);
+        } 
+        
+        if action is ObjectRetrievalHeaders {
+            httpMethod = GET;
+            populateGetObjectHeaders(requestHeaders, action);
+        } 
+        
+        if action is CREATE {
             httpMethod = PUT;
-            if action is ObjectCreationHeaders {
-                populateCreateObjectHeaders(requestHeaders, action);
-            }
         } else {
             httpMethod = GET;
-            if action is ObjectRetrievalHeaders {
-                populateGetObjectHeaders(requestHeaders, action);
-            }
         }
```

Comment:
Shall we use if - else if - else structure in here?

Reply:
I tried that,
getting the error `incompatible types: '("RETRIEVE"|"CREATE")' will not be matched to 'ballerinax/aws.s3:3.3.0:ObjectRetrievalHeaders'(BCE3603)` on 
```ballerina
if action is ObjectRetrievalHeaders {
```
but only on 2201.4.1

Reply:
Hmmm seems to be a bug. Okay lets proceed with this approach ftm